### PR TITLE
Add support for all dict-type editable attributes

### PR DESCRIPTION
**Pull request by dwright04**
This is to allow automated updates to the announcement banner.

### CODE REVIEW

**Comment by adammcmaster** [comment]
See also the overridden `__init__` and `set_raw`

**Comment by lcjohnso** [comment]
Expanding scope of this PR slightly: I'm targeting not only a fix for `Project.configuration` (as originally targeted by @dwright04), but also for `Workflow.retirement` and `Workflow.tasks` (two other prominent dict-type editable attributes).  I'll build directly from #222, expanding to additional dicts in `workflow.py` and adding solution to `project.py`.

**Comment by lcjohnso** [comment]
Commits are in.  This PR should now accomplish @dwright04's original goal as well as close #211 . 

@adammcmaster let me know what you think!

File: panoptes_client/workflow.py [hunk 1]
```diff
@@ -48,6 +61,10 @@ def save(self):
         """
```
Comment (adammcmaster) [inline panoptes_client/workflow.py:61]: Could you update the docstring here now that it's not just configuration that it's handling?

**Comment by lcjohnso** [comment]
Closing in favor of #251

**Pull request closed without merging**